src/main/java/org/veupathdb/eda/binaryfiles/dumper/ScanningBinaryMetadataProvider.java

```diff
@@ -0,0 +1,44 @@
+package org.veupathdb.eda.binaryfiles.dumper;
+
+import org.veupathdb.service.eda.ss.model.Entity;
+import org.veupathdb.service.eda.ss.model.Study;
+import org.veupathdb.service.eda.ss.model.db.FilteredResultFactory;
+import org.veupathdb.service.eda.ss.model.reducer.BinaryMetadataProvider;
+import org.veupathdb.service.eda.ss.model.tabular.TabularReportConfig;
+import org.veupathdb.service.eda.ss.model.variable.*;
+import org.veupathdb.service.eda.ss.model.variable.binary.EmptyBinaryProperties;
+
+import javax.sql.DataSource;
+import java.util.*;
+
+public class ScanningBinaryMetadataProvider implements BinaryMetadataProvider {
+    private Study study;
+    private DataSource dataSource;
+    private String schema;
+
+    public ScanningBinaryMetadataProvider(Study study, DataSource dataSource, String schema) {
+        this.study = study;
+        this.dataSource = dataSource;
+        this.schema = schema;
+    }
+
+    @Override
+    public Optional<BinaryProperties> getBinaryProperties(String studyAbbrev, Entity entity, String variableId) {
+        Entity fullyPopulatedEntity = study.getEntity(entity.getId())
+                .orElseThrow();
+        Variable var = fullyPopulatedEntity
+                .getVariable(variableId)
+                .orElseThrow();
+        if (var instanceof VariableWithValues) {
+            VariableWithValues varWithValues = (VariableWithValues) var;
```

Comment:
Consider moving this type-specific functionality into the actual types rather than switch/if-then on type.